### DISCUSSION TITLE
Default printer in DebugEnvironment?

### DISCUSSION THREAD

**Discussion by eimantas** [pointfreeco/swift-composable-architecture#709] [General]
Would you consider using `os_log` instead of `print` as default printer in DebugEnvironment?

https://github.com/pointfreeco/swift-composable-architecture/blob/67ccab6bce3b67d0c9e6d42919e7b245b6b14fa7/Sources/ComposableArchitecture/Debugging/ReducerDebugging.swift#L149-L152

**Comment by stephencelis** [comment]
We're definitely open to discussing improvements here! Do you have an override you use in your applications that you've been happy with?

We also wonder if os_log's configurability opens up any opportunities. While this is possible:

```swift
printer: @escaping (String) -> Void = { os_log("%@", $0) },
```

We're wondering if there's more we could/should do with the data passed along.

**Reply by virgilius-santos** [reply]
Hi! I need to save and send logs to our server
